logic/concludable/Concludable.java

```diff
@@ -120,15 +124,47 @@ static void copyIsaAndValues(ThingVariable oldOwner, ThingVariable newOwner) {
         copyValuesOntoVariable(oldOwner.value(), newOwner);
     }
 
-    static void copyLabelAndValueType(TypeVariable copyFrom, TypeVariable copyTo) {
+    static void copyLabelSubAndValueType(TypeVariable copyFrom, TypeVariable copyTo) {
         if (copyFrom.label().isPresent()) copyTo.label(Label.of(copyFrom.label().get().label()));
+        if (copyFrom.sub().isPresent()) {
+            SubConstraint subCopy = copyFrom.sub().get();
+            copyTo.sub(subCopy.type(), subCopy.isExplicit());
+            copyTo.sub().get().addHints(subCopy.typeHints());
+        }
         if (copyFrom.valueType().isPresent()) copyTo.valueType(copyFrom.valueType().get().valueType());
     }
 
     static TypeVariable copyVariableWithLabelAndValueType(TypeVariable copyFrom) {
         TypeVariable copy = TypeVariable.of(copyFrom.identifier());
-        copyLabelAndValueType(copyFrom, copy);
+        copyLabelSubAndValueType(copyFrom, copy);
         return copy;
     }
 
+    static boolean hasNoHints(Variable variable) {
+        //TODO: refactor once new 'all things' symbol is introduced
```

Comment:
this TODO is obsolete no?

Reply:
No, that will be implemented when my other PR is merged.